### DISCUSSION TITLE
string additions to externalized filenames dont update before files are created

### DISCUSSION THREAD

**Discussion by dprotter** [bonsai-rx/bonsai#1424] [Q&A]
Hi All! I've been trying to make a workflow that allows for quickly generating filenames that are updated with an animal number. That way one of our users can just update 1 node and then update all their associated filepaths in one go. The problem I have run into is that, while this works, most files are created as soon as I hit "run", but before the workflow has updated the filenames. If I run, stop, and run again everything looks correct. 

I've attached the workflow. 

Thanks!

Dave
[undergrad_GRAB_operant.zip](https://github.com/bonsai-rx/bonsai/files/11773584/undergrad_GRAB_operant.zip)

**Comment by bruno-f-cruz** [comment]
Check #1367.
